plugin/pkg/scheduler/schedulercache/cache.go

```diff
@@ -233,6 +233,44 @@ func (cache *schedulerCache) RemovePod(pod *api.Pod) error {
 	return nil
 }
 
+func (cache *schedulerCache) AddNode(node *api.Node) error {
+	cache.mu.Lock()
+	defer cache.mu.Unlock()
+
+	n, ok := cache.nodes[node.Name]
+	if !ok {
+		n = NewNodeInfo()
+		cache.nodes[node.Name] = n
+	}
+	return n.SetNode(node)
+}
+
+func (cache *schedulerCache) UpdateNode(oldNode, newNode *api.Node) error {
```

Comment:
> Sorry - I don't understand. Can you please clarify.

I would add a update function to avoid allocation by explicitly reusing the previous allocated object. I thought this update function is here to prepare for this purpose.